pyproject.toml

```diff
@@ -0,0 +1,117 @@
+[tool.poetry]
```

Comment:
@onuralpszr I am just taking a quick look as I am doing a more detailed review. (Looks awesome btw). First question:

Why poetry?

As a seasoned dev I dont care if it is popular or not but how would you convince me to use this? I prefer using official stuff generally TBH and I wonder why you chose poetry? Pls skip its popularity and sell it to me in a way because I want to understand why we should use it in this project. The official stuff had been doing my job already.

Reply:
I used poetry because I know how to use and handle plus I need something to handle "toml" file and my development setup. Poetry gives me "venv" generation and "poetry install" and all ready to use for me. After that "poetry shell" done. One big main reason is removal of "setup.py" and use "toml" and format is acceptable by official python because they have to comply that so that why.

Reply:
Forget the toml as you can use pyproject.toml without poetry right? pyproject.toml comes from a PEP but poetry doesnt. Since poetry is not a python official tool, I cannot rely on that project 100%. It is the same for click but click does sth related to business technicalities for me and if the official python offered me an easy cli tool I would not use click for example. I am giving the click example because you faced an exit code issue there for example. In the future we might face issues with poetry in other ways. 

I dont know much about poetry so pls let me investigate and research this a bit. the project is not a rocket to the moon but I am an addict of minimalism and consistency so let me check first:)

Reply:
For example take this project: https://github.com/ansible/anonymizer/blob/main/pyproject.toml

This also uses a pyproject.toml but uses the official setuptools as far as I understand. So using poetry feel like: while it has many many features, it is 3rd party and we could do the same with the good old setup tools.

Reply:
@onuralpszr I took a look at the general poetry stuff and I think I dont want to put yet another dependency to the project. Click was different (and as you can see we have very small count of deps currently)

Would you mind converting your pyproject.toml and the project in this PR to use the setuptools (like before)?

As I said before this project might be a good sample: https://github.com/ansible/anonymizer/blob/2f2e428d1bb9b1cd2950bec6ef3ba03639ee6e5f/pyproject.toml#L1 (it also uses tox but don't mind that, I am not ready for tox yet:)) )